connector/extensions/report-service/src/main/java/org/oaebudt/edc/report/ReportApiController.java

```diff
@@ -0,0 +1,150 @@
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.glassfish.jersey.media.multipart.FormDataParam;
+import org.oaebudt.edc.report.model.ReportType;
+import org.oaebudt.edc.report.repository.ReportStore;
+
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.Map;
+
+import static jakarta.ws.rs.core.MediaType.WILDCARD;
+
+@Path("/")
+@Consumes(WILDCARD)
+@Produces(WILDCARD)
+@MultipartConfig
+public class ReportApiController {
+
+    private final ReportStore reportStore;
+    private final String managementUrl;
+    private final Monitor monitor;
+
+
+    public ReportApiController(Monitor monitor, ReportStore reportStore, String managementUrl) {
+        this.reportStore = reportStore;
+        this.managementUrl = managementUrl;
+        this.monitor = monitor;
+    }
+
+    @POST
+    @Path("upload")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response uploadFile(@Context ContainerRequestContext requestContext, @FormDataParam("file") InputStream uploadedInputStream,
+                               @FormDataParam("title") String title,
+                               @FormDataParam("reportType") ReportType reportType) {
+
+        try (JsonReader reader = Json.createReader(uploadedInputStream)) {
+            JsonObject jsonObject = reader.readObject();
+
+            JsonObject jsonWithMeta = Json.createObjectBuilder(jsonObject)
+                    .add("title", title)
+                    .add("reportType", reportType.name())
+                    .build();
+
+            reportStore.saveReport(jsonWithMeta.toString(), reportType);
+
+            // create asset if it has not been created already
+            createAssetInConnector(title, reportType, requestContext.getHeaderString("Authorization"));
+
+            return Response.status(Response.Status.CREATED)
+                    .entity("{\"message\":\"JSON file uploaded and saved\"}")
+                    .build();
+        } catch (JsonException e) {
+            monitor.warning("Error uploading report", e);
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("{\"error\":\"Failed to process uploaded JSON\"}")
+                    .build();
+        } catch (Exception e) {
+            monitor.warning("Error uploading report", e);
+            return Response.serverError()
+                    .entity("{\"error\":\"Something went wrong\"}")
+                    .build();
+        }
+
+    }
+
+     void createAssetInConnector(String title, ReportType reportType, String authorization) {
+
+        var client = EdcConnectorClient.newBuilder()
+                .managementUrl(this.managementUrl)
+                .interceptor(builder -> builder.header("Authorization", authorization))
+                .build();
+
+        createDefaultPolicyDefinition(client, reportType);
+        createDefaultContractDefinition(client, reportType);
```

Comment:
It's not sane to create a new policy and a new contract definition for every report uploaded.
If policy is always the same (in this case, an empty one), it can be created during a seed phase.

Same for the contract definition, consider that it can target a set of asset determined by the `assetsSelector` field, so I can see a single Contract Definition that points to all the `TITLE_REPORT` and one to all the `ITEM_REPORT`, and, as for the policy, they can be created in a seed phase.